src/sdk.ts

```diff
@@ -1039,14 +1039,16 @@ export class OpenSeaSDK {
       throw new Error("Unsupported protocol");
     }
 
-    if (!order.clientSignature && order.orderHash) {
+    if (order.orderHash) {
       const result = await this.api.generateFulfillmentData(
         accountAddress,
         order.orderHash,
         order.protocolAddress,
         order.side
       );
-      order.clientSignature = result.fulfillment_data.orders[0].signature;
+      const signature = result.fulfillment_data.orders[0].signature;
+      order.clientSignature = signature;
+      order.protocolData.signature = signature;
```

Comment:
Can we pull this out to a separate function so it can be unit tested? A function which takes in accountAddress and order and outputs OrderWithCounter.

Reply:
@JoshuaSchmidt-OpenSea writing an integration test for fulfillment is a bit nuanced - you need to create a listing with one wallet and fulfill it with another